docs/discovering_thanos_components.md

```diff
@@ -0,0 +1,61 @@
+# Discovering Thanos Components
+
+Thanos is a set of components. In order to successfully run Thanos these components need to be able to talk to each other. 
```

Comment:
```suggestion
Service discovery is a vital place for Thanos components that allows to perform some logic against given set of APIs. Currently there are 2 places like this: 
```

Let's make this generic `SD` explanation - especially that part `components need to be able to talk to each other.` is extremely dangerous, because it indicates that gossip is actually needed ;p